server/server.go

```diff
@@ -41,17 +52,30 @@ func RegistrationFormHandler(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
-func RenderingRegistrationDetails(w http.ResponseWriter, r *http.Request) {
+func (s *Server) RenderingRegistrationDetails(w http.ResponseWriter, r *http.Request) {
 	registrationDetails := User{
-		Email:    r.FormValue("email"),
-		Username: r.FormValue("username"),
-		Password: r.FormValue("password"),
+		FirstName: r.FormValue("firstName"),
+		LastName:  r.FormValue("lastName"),
+		Email:     r.FormValue("email"),
+		Username:  r.FormValue("username"),
 	}
 
-	_ = registrationDetails
+	hashedPassword, err := HashPassword(r.FormValue("password"))
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+	registrationDetails.Password = hashedPassword
+
+	_, err = s.db.Query("INSERT INTO users (first_name, last_name, email, username, password) VALUES ($1, $2, $3, $4, $5)",
```

Comment:
For now it's okay to have this logic in the handler for the purposes of the PR, but we will talk soon about separating application logic a bit better to keep our handlers as simple as possible (trust me, this is what you want most of the time).